WorldProcessor prototype - version 0.0.1.0 (#2)
Max-Dov/strongson: WorldProcessor/README.md

> # 🏭 World Iterator

Reviewer: ```suggestion
# 🏭 World Processor
```
Везде называется как процессор, кроме названия

Author: Принято, добавлено.

> WebApi service that provide functionality to generate and iterate world.

Reviewer: ```suggestion
Web API service that provides functionality to generate and iterate world.
```

Author: Принято, добавлено.

> ## Usage

Reviewer: If it's Web API, better just call usage an API
```suggestion
## API
```

Author: Принято, добавлено.

> WorldProcessor have 2 endpoints:

Reviewer: ```suggestion
World Processor has 2 endpoints:
```

Author: Принято, добавлено.

> 1) **url:port/WorldProcessing/Generate** - action for generation initial world.

Reviewer: ```suggestion
### Generate World

`POST https://localhost:3000/WorldProcessing/Generate`

Request body:
```
* It is not numeric list (order is not important) and list heading should be headers, so it can be seeing in "heading" if readme had one and could be refered via link (e.g. "/readme.md#generate")

Author: Принято, добавлено.

	It accepts only POST request.
	request description:

> 	```

Reviewer: ```suggestion
	```language
```
If possible, make it language for nice synthax highlight, otherwise I recommend making model (aka "request description") TypeScript interface or JSON-line object as it's self explanatory and can be referenced via notes/contracts.md.

Author: Added typescript as language

	{
		"seed": => number,
		"dimensions" => Array<number>?
		"worldConfig": WorldConfig
	}
	```
2) **url:port/WorldProcessing/Iterate** - action for generating next world iteration
it accepts only POST request.
Request body description:
	```
	{
		"world": => World,
		"worldConfig": => WorldConfig
	}
	```

> ##  Compiled application

Reviewer: ```suggestion
##  Executable file
```
Optional task, but I think "compiled" and "application" is too broad description for simple ".exe" file.

Author: I prefer more "Latest application build", because it is not a single file

> Latest compiled application is located in `Release build` directory.

Reviewer: Make it link to github exact file on master branch, otherwise it doesn't mean latest, it means "this branch executable."

Author: Added link to application directory, becuase it is not just single file

> To use it, just run `WorldProcessor.WebApi.exe` executable file and find application address and port in console.

Reviewer: Do .dll and other files required for that .exe?
Is it possible to place everything within single .exe, so folder "build" for that branch would have simple single .exe file?

Author: It is possible to minimize final application, but some files like `appsettings.json` will stay due to application nature.

I will change current build to minimized

> ![CLI Preview](./readme-preview-editor.png "CLI Preview")

Reviewer: lmao, did you even preview your readme?

```suggestion
![CLI Preview](./readme-preview-cli.png "CLI Preview")
```

Also logs make no sense to me as screenshot has both :5000 and :5001 as ports, so what is actual port??

Can you like.. clean console when app starts and print simple message "World Processor started! Address: https://localhost:1234/"?

Author: Applicaiton listens both ports. 5000 for http and 5001 for https requests. This information will be added to readme file.
I don't know how to change default cli output) If it is important i can investigate this issue.
However image will be removed.

## Development
For development install .NET SDK - https://dotnet.microsoft.com/en-us/download.

> Use IDE that support development on .NET like [Visual Studio](https://visualstudio.microsoft.com/ru/) or [Intellij Rider](https://www.jetbrains.com/ru-ru/rider/).

Reviewer: Make it english links if readme is in english.

Author: Done

### Swagger

> To use Swagger functionality, get on `url:port/Swagger`.

Reviewer: Make that exact valid link, e.g.
```suggestion
To use Swagger functionality, get on `https://localhost:3000/Swagger`.
```
Or make port 3000 to be just "port", but default address would be nice as it is expected to be single variant as swagger would be used only for local development I guess.

Author: Added valid links

> Swagger is enabled only in **Debug** configuration.

Reviewer: And how to enable/disable that?

Author: To enable Swagger, just build application in Debug configuration, to disable - in Release configuration.
Application in build directory is built in Release configuration.
